static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -618,76 +726,103 @@ class RuleConditionsForm extends PureComponent<Props, State> {
-                          ),
-                        }}
-                        customInvalidTagMessage={item => {
-                          if (dataset !== Dataset.GENERIC_METRICS) {
-                            return null;
-                          }
-                          return (
-                            <SearchInvalidTag
-                              message={tct(
-                                "The field [field] isn't supported for performance alerts.",
-                                {
-                                  field: <code>{item.desc}</code>,
-                                }
-                              )}
-                              docLink="https://docs.sentry.io/product/alerts/create-alerts/metric-alert-config/#tags--properties"
-                            />
-                          );
-                        }}
-                        searchSource="alert_builder"
-                        defaultQuery={initialData?.query ?? ''}
-                        {...getSupportedAndOmittedTags(dataset, organization)}
-                        includeSessionTagsValues={dataset === Dataset.SESSIONS}
-                        disabled={disabled || isErrorMigration}
-                        useFormWrapper={false}
-                        organization={organization}
-                        placeholder={this.searchPlaceholder}
-                        onChange={onChange}
-                        query={initialData.query}
-                        // We only need strict validation for Transaction queries, everything else is fine
-                        highlightUnsupportedTags={
-                          organization.features.includes('alert-allow-indexed') ||
-                          (hasOnDemandMetricAlertFeature(organization) &&
-                            isOnDemandQueryString(value))
-                            ? false
-                            : dataset === Dataset.GENERIC_METRICS
-                        }
-                        onKeyDown={e => {
-                          /**
-                           * Do not allow enter key to submit the alerts form since it is unlikely
-                           * users will be ready to create the rule as this sits above required fields.
-                           */
-                          if (e.key === 'Enter') {
-                            e.preventDefault();
-                            e.stopPropagation();
+                      {organization.features.includes('search-query-builder-alerts') ? (
+                        <SearchQueryBuilder
+                          initialQuery={initialData?.query ?? ''}
+                          getTagValues={this.getEventFieldValues}
+                          placeholder={this.searchPlaceholder}
+                          searchSource="alert_builder"
+                          filterKeys={filterKeys}
```

Comment:
This file has a `getSupportedAndOmittedTags` function that it was using before. Does your new function do the same thing or do we will need that?

Reply:
Ah got it 👍 
yes, i'll add this to the filterKeys